### ISSUE TITLE
Toggle visible markers

### ISSUE THREAD

**Issue by lickybuay** [mehmetoguzderin/decorated_google_maps_flutter#1]
Could the markers toggle visibility? in the google_map_flutter is final visible true.

**Comment by mehmetoguzderin** [comment]
@lickybuay We will take a look at this soon!

**Comment by mehmetoguzderin** [comment]
> Could the markers toggle visibility? in the google_map_flutter is final visible true.

@lickybuay Do you want to change `final bool visible` of `Marker` after creating a `GoogleMap`?

**Comment by lickybuay** [comment]
Yes, the idea is toggle visibility of a marker, for example i have a list with checkbox, if checked marker visible, otherwise hide. 

Actually im making this changing position values on the list (array), i remove all markers and then remake the markers in the map again.

**Comment by mehmetoguzderin** [comment]
> Yes, the idea is toggle visibility of a marker, for example i have a list with checkbox, if checked marker visible, otherwise hide.
> 
> Actually im making this changing position values on the list (array), i remove all markers and then remake the markers in the map again.

@lickybuay That actually makes sense, if you opt for a list of `Marker`s, then you essentially have to recreate the objects at each occasion which could be avoided. Let me think about how to patch a way for that in our plugin! Meanwhile, don't hesitate to open other issues or pull requests about the plugin, appreciated!

**Comment by lickybuay** [comment]
i know that makes sense, but in google map you can make markers[indexOfMarker].setVisible(true or false); that don't remove the value of the list but toggle the visibility.

And if you check the doc of google_maps_flutter you will see that the marker have visible property, the thing is that it's final. 

`visible → bool
True if the marker is visible.`

https://pub.dev/documentation/google_maps_flutter/latest/google_maps_flutter/Marker-class.html

**Comment by mehmetoguzderin** [comment]
> i know that makes sense, but in google map you can make markers[indexOfMarker].setVisible(true or false); that don't remove the value of the list but toggle the visibility.
> 
> And if you check the doc of google_maps_flutter you will see that the marker have visible property, the thing is that it's final.
> 
> `visible → bool True if the marker is visible.`
> 
> https://pub.dev/documentation/google_maps_flutter/latest/google_maps_flutter/Marker-class.html

@lickybuay I meant what you say makes sense, editable visibility
